src/main/kotlin/no/fdk/dataservicecatalog/controller/ImportController.kt

```diff
@@ -0,0 +1,56 @@
+package no.fdk.dataservicecatalog.controller
+
+import no.fdk.dataservicecatalog.domain.ImportResult
+import no.fdk.dataservicecatalog.exception.OpenApiParseException
+import no.fdk.dataservicecatalog.handler.ImportHandler
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.http.ProblemDetail
+import org.springframework.http.ResponseEntity
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.*
+import java.net.URI
+
+@RestController
+@RequestMapping("/internal/catalogs/{catalogId}/import")
+class ImportController(private val importHandler: ImportHandler) {
+
+    @PreAuthorize(ADMIN)
+    @PostMapping(
+        consumes = [MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_YAML_VALUE],
+        produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    fun import(@PathVariable catalogId: String, @RequestBody dataService: String): ResponseEntity<Void> {
+        val importResult = importHandler.importOpenApi(catalogId, dataService)
+
+        return ResponseEntity
+            .created(URI("/internal/catalogs/${catalogId}/import/results/${importResult.id}"))
+            .build()
+    }
+
+    @PreAuthorize(ADMIN)
+    @GetMapping(value = ["/results"], produces = [MediaType.APPLICATION_JSON_VALUE])
+    fun result(@PathVariable catalogId: String): ResponseEntity<List<ImportResult>> {
+        return ResponseEntity.ok(importHandler.getResults(catalogId))
+    }
+
+    @PreAuthorize(ADMIN)
+    @GetMapping(value = ["/results/{id}"], produces = [MediaType.APPLICATION_JSON_VALUE])
+    fun result(@PathVariable catalogId: String, @PathVariable id: String): ResponseEntity<ImportResult> {
+        return importHandler.getResult(id)
+            ?.let { ResponseEntity.ok(it) }
+            ?: ResponseEntity(HttpStatus.NOT_FOUND)
+    }
```

Comment:
GET-endepunktene bør kanskje også være åpne for alle med lesetilgang?

Reply:
Det veit eg ikkje/er ikkje bestemt. Same rolle som i begrepskatalog nett no. Kan ta det opp med Kurt Stian og/eller Kjersti, men ikkje veldig viktig å få avklart nett no.